### ISSUE TITLE
amdgpu driver crash when using an RX 6700 XT

### ISSUE THREAD

**Issue by NomisIV** [NixOS/nixpkgs#122776]
**Describe the bug**
When upgrading to my new gpu, I wouldn't ba able to get any driver working.

**To Reproduce**
1. Buy an RX 6700 XT (good luck)
2. Use any of the mesa, amdgpu or ati video drivers
3. Use linuxPackages_latest
4. rebuild and restart

**Expected behavior**
It should preferrably just work

**Screenshots**
No screenshot, but relevant snippet from dmesg:
```
[    2.768311] amdgpu 0000:0b:00.0: Direct firmware load for amdgpu/navy_flounder_sos.bin failed with error -2
[    2.768314] amdgpu 0000:0b:00.0: amdgpu: failed to init sos firmware
[    2.768319] [drm:psp_sw_init [amdgpu]] *ERROR* Failed to load psp firmware!
[    2.768494] [drm:amdgpu_device_init.cold [amdgpu]] *ERROR* sw_init of IP block <psp> failed -2
[    2.768690] amdgpu 0000:0b:00.0: amdgpu: amdgpu_device_ip_init failed
[    2.768698] amdgpu 0000:0b:00.0: amdgpu: Fatal error during GPU init
[    2.768730] amdgpu: probe of 0000:0b:00.0 failed with error -2
```

**Additional context**
I have found [this archwiki post](https://bbs.archlinux.org/viewtopic.php?id=241528) highlighting the same issue but with a different graphics card.
It seems that the file `amdgpu/navy_flounder_sos.bin` should be included in the initramfs, if I understand correctly.

**Notify maintainers**

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.19, NixOS, 21.05.20210511.ca26ea5 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210503_6d2553a`
 - channels(root): `"home-manager, nixos-21.05pre286880.7cb76200088, nixos-hardware"`
 - channels(simon): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```
I'm using flakes, btw.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by TredwellGit** [comment]
Do you have `hardware.enableRedistributableFirmware = true;`?

**Comment by NomisIV** [comment]
I have not. I tried enabling it, but I didn't notice a difference

**Comment by NomisIV** [comment]
I have researched this a bit now, and it comes down to the graphics card not yet being supported by the driver. [See this post](https://bbs.archlinux.org/viewtopic.php?id=265263)

I am unsure with what needs to be done to fix this in NixOS

**Comment by jonringer** [comment]
> It seems that the file amdgpu/navy_flounder_sos.bin should be included in the initramfs, if I understand correctly.

Only files needed to mount the root file system really need to be in the initramfs. Later boot stages should be able to handle bringing this in.

**Comment by jonringer** [comment]
Also, there's been a `nixos-unstable` and a `nixpkgs-unstable` release since then. Updating your flake.lock should fix the issue.

**Comment by NomisIV** [comment]
I just tried rebuilding and things work now!
However I get some graphical glitches, possibly like mentioned in the original post

**Comment by vcunat** [comment]
This sounds solved to me; reopen otherwise.  Graphical glitches sound more likely to be a different problem, perhaps even upstream (i.e. distro-independent).

**Comment by NomisIV** [comment]
Yeah, it will hopefully be solved with https://github.com/NixOS/nixpkgs/pull/120325 or https://github.com/NixOS/nixpkgs/pull/119558

**Comment by vcunat** [comment]
Oh, then it's perhaps unfortunate that those won't make it to 21.05.  Perhaps some reliable override will be found (again), allowing to switch mesa versions more easily.

**Comment by jonringer** [comment]
#120325  can probably be backported, that's just a patch bump
